sdk/tables/Azure.Data.Tables/src/Queryable/ExpressionWriter.cs

```diff
@@ -155,9 +168,17 @@ private void VisitOperand(Expression e)
         {
             if (e is BinaryExpression || e is UnaryExpression)
             {
-                _builder.Append(UriHelper.LEFTPAREN);
-                Visit(e);
-                _builder.Append(UriHelper.RIGHTPAREN);
+                if (e is UnaryExpression unary && unary.NodeType == ExpressionType.TypeAs)
```

Comment:
Is there a functional benefit in dropping the parenthesis for casts?

Reply:
In this case the parens are not useful. The cast is only needed for the comparison to satisfy the compiler in that all dictionary entries are `object` so the writer is basically ignoring it. Leaving the parens here results in something like:
`(PartitionKey) eq "foo"`